attribution/token_perturbation.py

```diff
@@ -64,7 +83,7 @@ def get_most_similar_token_ids(
     # Find the nearest neighbor
     indices = nbrs.kneighbors(token_embedding.reshape(1, -1), return_distance=False)
 
-    return indices.flatten().tolist()
+    return indices.flatten().tolist()  # type: ignore
```

Comment:
This one seems to be because `kneighbors` returns `ndarray[Unknown, Unknown] | tuple[ndarray[Unknown, Unknown], ndarray[Unknown, Unknown]]`, so either an ndarray or a tuple of ndarrays.

The sklearn docs seems to say it is in fact a tuple so maybe their type definition is wrong: https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.NearestNeighbors.html#sklearn.neighbors.NearestNeighbors.kneighbors

One way to fix it is just to treat it explicitly like a tuple. The type error goes away if you do this:
```python
    indices, _ = nbrs.kneighbors(token_embedding.reshape(1, -1), return_distance=False)

    return indices.flatten().tolist()
```

Reply:
Fixed